### PR TITLE
Add autosuggest examples and amp-bind and action-impl changes it requires

### PR DESCRIPTION
**Pull request by cvializ**
Related-to #8825

### CODE REVIEW

File: extensions/amp-bind/0.1/bind-impl.js [hunk 1]
```diff
@@ -683,6 +683,10 @@ export class Bind {
   applyBinding_(boundProperty, element, newValue) {
     const property = boundProperty.property;
     switch (property) {
+      case 'value':
```
Comment (kmh287) [inline extensions/amp-bind/0.1/bind-impl.js]: [Bind validator should already support binding to the `value` attribute on input elements](https://github.com/ampproject/amphtml/blob/master/extensions/amp-bind/0.1/bind-validator.js#L323). If this code is removed, does your example break?

**Comment by cvializ** [comment]
Should this be renamed to `autosuggest example` instead? I feel like that might communicate our intent a little better, especially for an ABE sample.

File: examples/autocomplete.amp.html [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        [value]="query"
+        autofocus/>
+      <button class="search-submit" type="submit">Search</button>
+    </div>
+    <amp-list
+      class="autocomplete-box"
+      layout="fixed-height"
+      height="120"
+      src="/form/autocomplete/query"
+      [src]="'/form/autocomplete/query?q=' + autocomplete"
```
Comment (kmh287) [inline examples/autocomplete.amp.html]: If `autocomplete` isn't defined in the first `AMP.setState` action, the src attribute will be changed to `'/form/autocomplete/query?q=null"`

I recommend changing this to

```HTML
 [src]="'/form/autocomplete/query' + (autocomplete ? '?q=' + autocomplete : '')
```

that way, null can't get into the query param and the default evaluation of the expression for `[src]` is equal to the initial vlaue for `src`.

File: extensions/amp-bind/0.1/bind-impl.js (hunk 1 above)
Comment (cvializ) [inline extensions/amp-bind/0.1/bind-impl.js]: It sets the attribute, but that only works the first time. Subsequent changes to the value attribute don't change the text inside the element : /

File: examples/autocomplete.amp.html (hunk 2 above)
Comment (cvializ) [inline examples/autocomplete.amp.html]: Good catch, done.

File: build-system/app.js [hunk 3]
```diff
@@ -214,6 +214,40 @@ app.use('/form/search-json/get', function(req, res) {
   });
 });
 
+const autocompleteLanguages = ['ActionScript', 'AppleScript', 'Asp', 'BASIC',
```
Comment (aghassemi) [inline build-system/app.js]: list is missing my favourite languages, will not approve PR.

File: examples/autocomplete.amp.html [hunk 4]
```diff
@@ -0,0 +1,118 @@
+<!doctype html>
```
Comment (aghassemi) [inline examples/autocomplete.amp.html]: plz run through validator to be sure valid then add to `test-example-validation`

File: build-system/app.js (hunk 3 above)
Comment (cvializ) [inline build-system/app.js]: hahaha too bad!

**Comment by kmh287** [comment]
a little late but LGTM :sweat_smile:

**Pull request merged**